tests/resources/generator/validate_units_in_model.cellml

```diff
@@ -0,0 +1,232 @@
+                <apply>
+                    <log/>
+                    <logbase>
+                        <cn cellml:units="dimensionless">2</cn>
+                    </logbase>
+                    <ci>in_A</ci>
+                </apply>
+            </apply>
+            <!-- Not allowed to have a mismatch in the base and actual logarithm-->
+            <apply>
+                <eq/>
+                <ci>out_B</ci>
+                <apply>
+                    <log/>
+                    <logbase>
+                        <cn cellml:units="second">2</cn>
+                    </logbase>
+                    <ci>in_A</ci>
+                </apply>
+            </apply>
+            <!-- Allowed-->
+            <apply>
+                <eq/>
+                <ci>out_C</ci>
+                <apply>
+                    <log/>
+                    <logbase>
+                        <cn cellml:units="dimensionless">2</cn>
+                    </logbase>
+                    <ci>in_B</ci>
+                </apply>
+            </apply>
+        </math>
+        <!-- Allowed-->
+    </component>
+    <component name="power_root_operators">
+        <variable interface="public" name="f" units="volt_per_farad"/>
+        <variable interface="public" name="g" units="volt_per_farad"/>
+        <variable interface="public" name="h" units="volt_per_farad"/>
+        <variable interface="public" name="i" units="dimensionless"/>
+        <variable interface="public" name="j" units="dimensionless"/>
+        <math xmlns="http://www.w3.org/1998/Math/MathML">
+            <apply>
+                <eq/>
+                <ci>g</ci>
+                <apply>
+                    <root/>
+                    <apply>
+                        <power/>
+                        <ci>f</ci>
+                        <cn cellml:units="dimensionless">2</cn>
+                    </apply>
+                </apply>
+            </apply>
+            <!-- Allowed-->
+            <apply>
+                <eq/>
+                <ci>h</ci>
+                <apply>
+                    <root/>
+                    <apply>
+                        <power/>
+                        <ci>f</ci>
+                        <cn cellml:units="dimensionless">2</cn>
+                    </apply>
+                </apply>
+            </apply>
+            <!-- Allowed-->
+            <apply>
+                <eq/>
+                <ci>j</ci>
+                <apply>
+                    <exp/>
+                    <ci>i</ci>
+                </apply>
+            </apply>
+            <!-- Allowed as both e and i are dimensionless although i is not a fixed value.-->
+            <apply>
+                <eq/>
+                <ci>j</ci>
+                <apply>
+                    <power/>
+                    <ci>h</ci>
+                    <cn cellml:units="second">2</cn>
+                </apply>
+            </apply>
+        </math>
+        <!-- Not allowed - 2 is not dimensionless-->
```

Comment:
I will have to double check, but I think in some of the other models used as unit tests which @agarny put in (perhaps in the coverage test as well) there are instances of variables used as powers.

Reply:
Indeed, the coverage test (at least) has instances of variables used as powers ([here](https://github.com/cellml/libcellml/blob/7e991e0f69d2512576659d1c41ef068ae95f7c27/tests/resources/generator/coverage/model.cellml#L396-L412) for example).

Reply:
Are we even allowed to have dimensional stuff as an exponent?  How on earth do the units handle that?

Reply:
As far as I can remember, only a `dimensionless` exponent makes sense. This is the reason `n` in `m^n` is `dimensionless`.

Reply:
Yep, @kerimoyle it doesn't make sense to have dimensional stuff as exponents (obviously), but to the best of my recollection there is nothing preventing someone from writing a CellML file intentionally with exponents which are not dimensionless, hence why it is still good to check if it is in case someone's messed up the CellML or incorrectly defined a variable with units.